api/src/utils/encryption/index.ts

```diff
@@ -2,21 +2,24 @@ import * as crypto from 'crypto';
 import * as ed from 'noble-ed25519';
 import * as bs58 from 'bs58';
 
-export const createChallenge = (): string => {
-	const challenge = crypto.randomBytes(30).toString('hex');
-	return crypto.createHash('sha256').update(challenge).digest().toString();
+export const createNonce = (): string => {
+	return crypto.randomBytes(20).toString('hex');
 };
 
 export const getHexEncodedKey = (base58Key: string): string => {
 	return bs58.decode(base58Key).toString('hex');
 };
 
-export const signChallenge = async (privateKey: string, challenge: string): Promise<string> => {
-	return await ed.sign(challenge, privateKey);
+const hashNonce = (nonce: string) => crypto.createHash('sha256').update(nonce).digest().toString();
+
+export const signNonce = async (privateKey: string, nonce: string): Promise<string> => {
+	const hash = hashNonce(nonce);
+	return await ed.sign(hash, privateKey);
 };
 
-export const verifiyChallenge = async (publicKey: string, challenge: string, signature: string): Promise<boolean> => {
-	return await ed.verify(signature, challenge, publicKey);
+export const verifySignedNonce = async (publicKey: string, nonce: string, signature: string): Promise<boolean> => {
```

Comment:
Here as well, length of the parameters should be more or less the same in all cases, so I would also check it

Reply:
ha ok I had this but removed it because I was unclear if I should to do so :P